### PR TITLE
Allows the proxy settings selection trough the environment variables.

### PR DESCRIPTION
**Pull request by kozmaadrian**
## Description

Allows the proxy settings selection trough the environment variables.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by godanny86** [comment]
Nice @kozmaadrian ! Do you mind updating the Readme w/ this PR as well? https://github.com/adobe/aem-guides-wknd-graphql/edit/main/react-app/README.md

**Comment by kozmaadrian** [comment]
@godanny86 Thanks for your review. I've added the missing auth method exports and updated the Readme.

**Comment by kozmaadrian** [comment]
Hi @godanny86, please let me know if there's anything else to add to this PR. Thanks.

**Comment by godanny86** [comment]
@kozmaadrian can you sign the Adobe Open Source CLA? https://opensource.adobe.com/cla.html

**Comment by godanny86** [comment]
Oh never mind. Disregard my previous comment

File: react-app/.env.development [hunk 1]
```diff
@@ -4,7 +4,9 @@
 REACT_APP_HOST_URI=http://localhost:4502
 REACT_APP_GRAPHQL_ENDPOINT=/content/graphql/global/endpoint.json
 #AUTH (Choose one method)
-# For Bearer auth, use DEV token from Cloud console
+# Authentication methods: 'service-token', 'dev-token', 'basic'
```
Comment (godanny86) [inline react-app/.env.development:7]:
```suggestion
# Authentication methods: 'service-token', 'dev-token', 'basic' or leave blank to use no authentication
```

File: react-app/README.md [hunk 2]
```diff
@@ -44,6 +44,10 @@ Several [environment variables](https://create-react-app.dev/docs/adding-custom-
 
 * `REACT_APP_HOST_URI=http://localhost:4502` - Set to AEM target host
 * `REACT_APP_GRAPHQL_ENDPOINT=/content/graphql/global/endpoint.json` - Set the GraphQL endpoint path
+* `REACT_APP_AUTH_METHOD=` - The preferred authentication method. Optional, as per default no authentication is used.
+  * `service-token` - use Service token exchange for Cloud Env PROD
+  * `dev-token` - use Dev token for local development with Cloud Env
+  * `basic` - use user/pass for local development with Local Author Env
```
Comment (godanny86) [inline react-app/README.md:50]:
```suggestion
  * `basic` - use user/pass for local development with Local Author Env
  * leave blank to use no authentication method
```

**Review by godanny86 — APPROVED** [review]

**Pull request merged**